src/app/events/[slug]/components/EventsSection.tsx

```diff
@@ -1,18 +1,47 @@
+import { z } from 'zod'
+
+import { Card } from '@/components/Card'
+import { CardGrid } from '@/components/CardGrid'
 import { PageSection } from '@/components/PageSection'
 
 import type { Event } from '../../types/eventType'
+import { getMetaData } from '../../utils/getMetaData'
+import { getLumaEvents } from '../services/lumaService'
+import { mapLumaEventToMetadata } from '../utils/mapLumaEventToMetadata'
 
 type EventsSectionProps = NonNullable<Event['lumaEventsSection']>
 
-export function EventsSection({ title, embedLink }: EventsSectionProps) {
+export async function EventsSection({ title, embedLink }: EventsSectionProps) {
+  const events = await getLumaEvents()
+
   return (
     <PageSection kicker="Explore" title={title || 'Events'}>
-      <iframe
-        src={embedLink}
-        width="100%"
-        height="720"
-        className="rounded-lg"
-      ></iframe>
+      {events.length > 0 ? (
+        <CardGrid cols="smTwo">
+          {events.map(({ event }) => (
+            <Card
+              key={event.api_id}
+              title={event.name}
+              description={event.description}
+              metaData={getMetaData(mapLumaEventToMetadata(event))}
+              textIsClamped={true}
+              cta={{
+                text: 'View Event Details',
+                href: event.url,
+              }}
+            />
+          ))}
+        </CardGrid>
+      ) : embedLink ? (
```

Comment:
Not a big fan of double ternary operators, I think we can do better than that :)

The pattern we usually follow is we use `try catch` in the component fetching the data:
if success -> render the list
if success but data is empty -> show empty state
if error - offer a fallback (the iframe in our case)